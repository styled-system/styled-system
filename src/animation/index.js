import { system } from '../core'

const config = {
  animation: {
    property: 'animation',
    scale: 'animations',
  },
  animationName: {
    property: 'animationName',
    scale: 'animationNames',
  },
  animationDuration: {
    property: 'animationDuration',
    scale: 'animationDurations',
  },
  animationTimingFunction: {
    property: 'animationTimingFunction',
    scale: 'animationTimingFunctions',
  },
  animationDelay: {
    property: 'animationDelay',
    scale: 'animationDelays',
  },
  animationDirection: {
    property: 'animationDirection',
    scale: 'animationDirections',
  },
  animationIterationCount: {
    property: 'animationIterationCount',
    scale: 'animationIterationCounts',
  },
  animationFillMode: {
    property: 'animationFillMode',
    scale: 'animationFillModes',
  },
  animationPlayState: {
    property: 'animationPlayState',
    scale: 'animationPlayStates',
  },
}

export const animation = system(config)

export default animation
