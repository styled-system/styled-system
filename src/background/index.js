import { system } from '../core'

export const backgroundConfig = {
  background: true,
  backgroundImage: true,
  backgroundSize: true,
  backgroundPosition: true,
  backgroundRepeat: true,
}

backgroundConfig.bgImage = backgroundConfig.backgroundImage
backgroundConfig.bgSize = backgroundConfig.backgroundSize
backgroundConfig.bgPosition = backgroundConfig.backgroundPosition
backgroundConfig.bgRepeat = backgroundConfig.backgroundRepeat

export const background = system(backgroundConfig)

export default background
