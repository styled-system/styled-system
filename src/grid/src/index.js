import { system } from '../core'

const defaults = {
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
}

const config = {
  gridGap: {
    property: 'gridGap',
    scale: 'space',
    defaultScale: defaults.space,
  },
  gridColumnGap: {
    property: 'gridColumnGap',
    scale: 'space',
    defaultScale: defaults.space,
  },
  gridRowGap: {
    property: 'gridRowGap',
    scale: 'space',
    defaultScale: defaults.space,
  },
  gridColumn: true,
  gridRow: true,
  gridAutoFlow: true,
  gridAutoColumns: true,
  gridAutoRows: true,
  gridTemplateColumns: true,
  gridTemplateRows: true,
  gridTemplateAreas: true,
  gridArea: true,
}

export const grid = system(config)
export default grid
