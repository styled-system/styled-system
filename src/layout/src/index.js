import { system, get } from '../core'

const isNumber = n => typeof n === 'number' && !isNaN(n)
const getWidth = (n, scale) =>
  get(scale, n, !isNumber(n) || n > 1 ? n : n * 100 + '%')

const config = {
  width: {
    property: 'width',
    scale: 'sizes',
    transform: getWidth,
  },
  height: {
    property: 'height',
    scale: 'sizes',
  },
  minWidth: {
    property: 'minWidth',
    scale: 'sizes',
  },
  minHeight: {
    property: 'minHeight',
    scale: 'sizes',
  },
  maxWidth: {
    property: 'maxWidth',
    scale: 'sizes',
  },
  maxHeight: {
    property: 'maxHeight',
    scale: 'sizes',
  },
  size: {
    properties: ['width', 'height'],
    scale: 'sizes',
  },
  overflow: true,
  overflowX: true,
  overflowY: true,
  display: true,
  verticalAlign: true,
}

export const layout = system(config)
export default layout
