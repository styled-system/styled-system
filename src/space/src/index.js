import { get, system, compose } from '../core'

const defaults = {
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
}

const isNumber = n => typeof n === 'number' && !isNaN(n)

const getMargin = (n, scale) => {
  if (!isNumber(n)) {
    return get(scale, n, n)
  }

  const isNegative = n < 0
  const absolute = Math.abs(n)
  const value = get(scale, absolute, absolute)
  if (!isNumber(value)) {
    return isNegative ? '-' + value : value
  }
  return value * (isNegative ? -1 : 1)
}

const configs = {}
configs.margin = {
  margin: {
    property: 'margin',
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginTop: {
    property: 'marginTop',
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginRight: {
    property: 'marginRight',
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginBottom: {
    property: 'marginBottom',
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginLeft: {
    property: 'marginLeft',
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginX: {
    properties: ['marginLeft', 'marginRight'],
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
  marginY: {
    properties: ['marginTop', 'marginBottom'],
    scale: 'space',
    transform: getMargin,
    defaultScale: defaults.space,
  },
}
configs.margin.m = configs.margin.margin
configs.margin.mt = configs.margin.marginTop
configs.margin.mr = configs.margin.marginRight
configs.margin.mb = configs.margin.marginBottom
configs.margin.ml = configs.margin.marginLeft
configs.margin.mx = configs.margin.marginX
configs.margin.my = configs.margin.marginY

configs.padding = {
  padding: {
    property: 'padding',
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingTop: {
    property: 'paddingTop',
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingRight: {
    property: 'paddingRight',
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingBottom: {
    property: 'paddingBottom',
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingLeft: {
    property: 'paddingLeft',
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingX: {
    properties: ['paddingLeft', 'paddingRight'],
    scale: 'space',
    defaultScale: defaults.space,
  },
  paddingY: {
    properties: ['paddingTop', 'paddingBottom'],
    scale: 'space',
    defaultScale: defaults.space,
  },
}
configs.padding.p = configs.padding.padding
configs.padding.pt = configs.padding.paddingTop
configs.padding.pr = configs.padding.paddingRight
configs.padding.pb = configs.padding.paddingBottom
configs.padding.pl = configs.padding.paddingLeft
configs.padding.px = configs.padding.paddingX
configs.padding.py = configs.padding.paddingY

export const margin = system(configs.margin)
export const padding = system(configs.padding)
export const space = compose(margin, padding)

export default space
