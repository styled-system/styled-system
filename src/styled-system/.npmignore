src
test
babel.config.js
