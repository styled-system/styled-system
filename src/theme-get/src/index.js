import { get } from '../core'

export const themeGet = (path, fallback = null) => props =>
  get(props.theme, path, fallback)
export default themeGet
