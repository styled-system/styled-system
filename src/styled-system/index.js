import { createStyleFunction, createParser } from '../core';
// v4 api shims
import layout from '../layout';
import space from '../space';
import color from '../color';
import typography from '../typography';
import flexbox from '../flexbox';
import grid from '../grid';
import border from '../border';
import background from '../background';
import position from '../position';
import animation from '../animation';
import listStyle from '../list-style';

export { get, createParser, createStyleFunction, compose, system } from '../core';

export { space } from '../space';
export { color } from '../color';
export { layout } from '../layout';
export { typography } from '../typography';
export { flexbox } from '../flexbox';
export { border } from '../border';
export { background } from '../background';
export { position } from '../position';
export { grid } from '../grid';
export { shadow } from '../shadow';
export { css } from '../css';
export { cursor } from '../cursor';
export { animation } from '../animation';
export { objectFit } from '../object-fit';
export { listStyle } from '../list-style';
export { default as boxShadow, default as textShadow } from '../shadow';

export { variant, buttonStyle, textStyle, colorStyle } from '../variant';

const {
  padding,
  paddingTop,
  paddingRight,
  paddingBottom,
  paddingLeft,
  paddingX,
  paddingY,
  margin,
  marginTop,
  marginRight,
  marginBottom,
  marginLeft,
  marginX,
  marginY,
} = space;
const {
  width,
  height,
  minWidth,
  minHeight,
  maxWidth,
  maxHeight,
  size,
  verticalAlign,
  display,
  overflow,
  overflowX,
  overflowY,
} = layout;
const { textColor, opacity, backgroundColor, fill: fillColor } = color;
const {
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  letterSpacing,
  textIndent,
  textTransform,
  textDecoration,
  whiteSpace,
  textAlign,
  fontStyle,
} = typography;

const {
  alignItems,
  alignContent,
  justifyItems,
  justifyContent,
  flexWrap,
  flexDirection,
  flex,
  flexGrow,
  flexShrink,
  flexBasis,
  justifySelf,
  alignSelf,
  order,
} = flexbox;
const {
  gridGap,
  gridColumnGap,
  gridRowGap,
  gridColumn,
  gridRow,
  gridAutoFlow,
  gridAutoColumns,
  gridAutoRows,
  gridTemplateColumns,
  gridTemplateRows,
  gridTemplateAreas,
  gridArea,
} = grid;
const {
  borderWidth,
  borderStyle,
  borderColor,
  borderTop,
  borderRight,
  borderBottom,
  borderLeft,
  borderRadius,
} = border;
const { backgroundImage, backgroundSize, backgroundPosition, backgroundRepeat } = background;
const { zIndex, top, right, bottom, left } = position;
const {
  animationName,
  animationDuration,
  animationTimingFunction,
  animationDelay,
  animationDirection,
  animationIterationCount,
  animationFillMode,
  animationPlayState,
} = animation;
const { listStyleImage, listStylePosition, listStyleType } = listStyle;
export { default as borders } from '../border';
export {
  // space
  padding,
  paddingTop,
  paddingRight,
  paddingBottom,
  paddingLeft,
  paddingX,
  paddingY,
  margin,
  marginTop,
  marginRight,
  marginBottom,
  marginLeft,
  marginX,
  marginY,
  // layout
  width,
  height,
  minWidth,
  minHeight,
  maxWidth,
  maxHeight,
  size,
  verticalAlign,
  display,
  overflow,
  overflowX,
  overflowY,
  // color
  textColor,
  backgroundColor,
  opacity,
  fillColor,
  // typography
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  letterSpacing,
  textIndent,
  textTransform,
  textDecoration,
  whiteSpace,
  textAlign,
  fontStyle,
  // flexbox
  alignItems,
  alignContent,
  justifyItems,
  justifyContent,
  flexWrap,
  flexDirection,
  flex,
  flexGrow,
  flexShrink,
  flexBasis,
  justifySelf,
  alignSelf,
  order,
  // grid
  gridGap,
  gridColumnGap,
  gridRowGap,
  gridColumn,
  gridRow,
  gridAutoFlow,
  gridAutoColumns,
  gridAutoRows,
  gridTemplateColumns,
  gridTemplateRows,
  gridTemplateAreas,
  gridArea,
  // border
  borderWidth,
  borderStyle,
  borderColor,
  borderTop,
  borderRight,
  borderBottom,
  borderLeft,
  borderRadius,
  // background
  backgroundImage,
  backgroundSize,
  backgroundPosition,
  backgroundRepeat,
  // position
  zIndex,
  top,
  right,
  bottom,
  left,
  // animation
  animationName,
  animationDuration,
  animationTimingFunction,
  animationDelay,
  animationDirection,
  animationIterationCount,
  animationFillMode,
  animationPlayState,
  // list style
  listStyleImage,
  listStylePosition,
  listStyleType,
};

// v4 style API shim
export const style = ({
  prop,
  cssProperty,
  alias,
  key,
  transformValue,
  scale,
  // new api
  properties,
}) => {
  const config = {};
  config[prop] = createStyleFunction({
    properties,
    property: cssProperty || prop,
    scale: key,
    defaultScale: scale,
    transform: transformValue,
  });
  if (alias) config[alias] = config[prop];
  const parse = createParser(config);

  return parse;
};
