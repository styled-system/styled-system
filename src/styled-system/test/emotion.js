import React from 'react'
import renderer from 'react-test-renderer'
import serializer, { matchers } from 'jest-emotion'
import styled from '@emotion/styled'
import { space } from '../src'

expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

const render = el => renderer.create(el).toJSON()

describe('emotion', () => {
  test('default props can be overridden', () => {
    const Box = styled('div')(space)
    Box.defaultProps = {
      px: 0,
      py: 0
    }
    const json = render(<Box px={2} py={1} />)
    expect(json).toHaveStyleRule('padding-left', '8px')
    expect(json).toHaveStyleRule('padding-right', '8px')
    expect(json).toHaveStyleRule('padding-top', '4px')
    expect(json).toHaveStyleRule('padding-bottom', '4px')
  })

  test('responsive styles are rendered in the correct order', () => {
    const Box = styled('div')(space)
    const json = render(
      <Box
        m={[1, null, 3]}
        p={[1, 2, 3]}
      />
    )
    expect(json).toMatchSnapshot()
  })

  test('object breakpoints work with defaultProps', () => {
    const Box = styled('div')(space)
    const theme = {
      disableStyledSystemCache: true,
      breakpoints: {
        small: '32em',
        medium: '40em',
      }
    }
    Box.defaultProps = {
      theme,
      p: {
        _: 0,
        medium: 4,
      },
      m: {
        small: 2,
        medium: 3,
      }
    }
    const json = render(
      <Box />
    )
    expect(json).toMatchSnapshot()
  })
})

