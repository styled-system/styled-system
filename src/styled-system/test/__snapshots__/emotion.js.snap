// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emotion object breakpoints work with defaultProps 1`] = `
.emotion-0 {
  padding: 0;
}

@media screen and (min-width:32em) {
  .emotion-0 {
    margin: 8px;
  }
}

@media screen and (min-width:40em) {
  .emotion-0 {
    padding: 32px;
    margin: 16px;
  }
}

<div
  className="emotion-0"
/>
`;

exports[`emotion responsive styles are rendered in the correct order 1`] = `
.emotion-0 {
  margin: 4px;
  padding: 4px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    padding: 8px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    margin: 16px;
    padding: 16px;
  }
}

<div
  className="emotion-0"
/>
`;
